sound/soc/sof/intel/hda-dai.c

```diff
@@ -617,12 +617,6 @@ static int hda_dai_suspend(struct hdac_bus *bus)
 			sdai = swidget->private;
 			ops = sdai->platform_private;
 
-			ret = hda_link_dma_cleanup(hext_stream->link_substream,
-						   hext_stream,
-						   cpu_dai);
-			if (ret < 0)
-				return ret;
-
```

Comment:
This seems to contradict the comments line 599-602, and it's not clear why this causes a hard lockup?

Reply:
Yeah, something is not right for sure, but it prevents the lockup.
If we suspend with active audio we run through the pre_trigger/trigger/post_trigger sequence (and SUSPEND/STOP/PAUSE_PUSH is identical up to post_trigger) then we call hda_link_dma_cleanup().

With the suspend while paused we just have a longer delay between the triggers and the hda_link_dma_cleanup()

But, we have failing FE triggers, so we might fail to stop host side and this has some side effect on the link side if it is cleaned?
We cannot see this as the system is hard locked..